src/ui/tabpopup.c

```diff
@@ -378,28 +379,59 @@ meta_ui_tab_popup_new (const MetaTabEntry *entries,
           tmp = tmp->next;
 
           ++left;
-          ++right;
         }
 
       ++top;
-      ++bottom;
     }
 
   /* remove all the temporary text */
   gtk_label_set_text (GTK_LABEL (popup->label), "");
   /* Make it so that we ellipsize if the text is too long */
   gtk_label_set_ellipsize (GTK_LABEL (popup->label), PANGO_ELLIPSIZE_END);
 
-  /* Limit the window size to no bigger than screen_width/4 */
-  if (max_label_width>(screen_width/4))
+  int default_window_width = 0; /* 0 == small as possible, truncate label */
+
+  if (consider_label_width && top <= 1 && left < width)
     {
-      max_label_width = screen_width/4;
-    }
```

Comment:
I wonder if we can somehow limit the max width to a fraction of the screen, say, 80% or 90% of it, and automatically wrap - so basically if the widget is too large for the user's screen, instead of honoring their setting, we add a new row instead.

Reply:
This could also be possible, but I very much like the feature to configure the max number columns myself. So a solution could be to allow max-columns = 0 and this could mean: 80% of screen size.

Reply:
IMHO in general it should be possible to let the user limit the size of the popup and let not the screen determine this limit: The label text in the popup is centered. The user could have problems to read the label while switching if the screen width is very large and therefore the distance from the left border of the popup to the beginning of the label text could become also very large.